pkg/logs/input/docker/tailer.go

```diff
@@ -134,13 +137,21 @@ func (t *Tailer) readForever() {
 		default:
 			inBuf := make([]byte, 4096)
 			n, err := t.reader.Read(inBuf)
-			if err != nil {
-				// an error occurred, stop from reading new logs
-				if err != io.EOF {
+			if err != nil { // an error occurred, stop from reading new logs
+				switch {
+				case isClosedConnError(err):
+					// This error is raised when the agent is stopping
+					return
+				case err == io.EOF:
+					// This error is raised when the container is stopping
+					t.source.RemoveInput(t.ContainerID)
+					return
+				default:
 					t.source.Status.Error(err)
-					log.Error("Err: ", err)
+					log.Errorf("Error reading logs for container %v: %v", ShortContainerID(t.ContainerID), err)
```

Comment:
This is very nitpicky again, but I feel like our log messages are not very consistent both in vocabulary and grammar. It's probably not that important but I think having the same message pattern across a package helps with debugging and clarity. Here I've seen a bunch of "tailing" vs "reading logs", could we choose one and stick with it ? Same with "error" vs "could not".